metricbeat/helper/server/http/http.go

```diff
@@ -66,21 +73,32 @@ func NewHttpServer(mb mb.BaseMetricSet) (server.Server, error) {
 	}
 
 	httpServer := &http.Server{
-		Addr:    fmt.Sprintf("%s:%d", config.Host, config.Port),
+		Addr:    net.JoinHostPort(config.Host, strconv.Itoa(int(config.Port))),
 		Handler: http.HandlerFunc(h.handleFunc),
 	}
+	if tlsConfig != nil {
+		httpServer.TLSConfig = tlsConfig.BuildModuleConfig(config.Host)
+	}
 	h.server = httpServer
 
 	return h, nil
 }
 
 func (h *HttpServer) Start() error {
 	go func() {
-
-		logp.Info("Starting http server on %s", h.server.Addr)
-		err := h.server.ListenAndServe()
-		if err != nil && err != http.ErrServerClosed {
-			logp.Critical("Unable to start HTTP server due to error: %v", err)
+		if h.server.TLSConfig != nil {
+			logp.Info("Starting HTTPS server on %s", h.server.Addr)
```

Comment:
Is it HTTP here?

Reply:
Hi Kaiyan, is HTTPS.

Reply:
Hummmm... there is this logp.Critical that is not adjusted.. thanks i will change it now